pkg/cmd/admin/node/evacuate.go

```diff
@@ -0,0 +1,156 @@
+package node
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	kerrors "github.com/GoogleCloudPlatform/kubernetes/pkg/util/errors"
+)
+
+type EvacuateOptions struct {
+	Options *NodeOptions
+
+	// Optional params
+	DryRun bool
+	Force  bool
+}
+
+func (e *EvacuateOptions) AddFlags(cmd *cobra.Command) {
+	flags := cmd.Flags()
+
+	flags.BoolVar(&e.DryRun, "dry-run", false, "Show pods that will be migrated. Optional param for --evacuate")
+	flags.BoolVar(&e.Force, "force", false, "Delete pods not backed by replication controller. Optional param for --evacuate")
+}
+
+func (e *EvacuateOptions) Run() error {
+	r, err := e.Options.GetNodeResource()
+	if err != nil {
+		return err
+	}
+
+	errList := []error{}
+	nodeCount := 0
+	err = r.Visit(func(info *resource.Info) error {
+		nodeCount++
+		node, ok := info.Object.(*kapi.Node)
+		if !ok {
+			err = fmt.Errorf("cannot convert input to Node: ", reflect.TypeOf(info.Object))
+			errList = append(errList, err)
+			// Don't bail out if one node fails
+			return nil
+		}
+
+		err := e.RunEvacuate(node)
+		if err != nil {
+			errList = append(errList, err)
+			// Don't bail out if one node fails
+			return nil
+		}
+		return nil
+	})
+	if len(errList) != 0 {
+		return kerrors.NewAggregate(errList)
+	}
+	if nodeCount == 0 {
+		if len(e.Options.NodeNames) > 0 {
```

Comment:
Seems like this condition should be different.  If any of the requested nodes were not found, list them.  If I requested two and you found one, I wouldn't get this error.